app/services/katello/pulp3/repository_mirror.rb

```diff
@@ -171,8 +171,13 @@ def publication_options(repository_version)
 
       def create_publication
         if (href = version_href)
-          publication_data = api.publication_class.new(publication_options(href))
-          api.publications_api.create(publication_data)
+          if repo_service.repo.content_type == "deb" && !smart_proxy.pulp_primary?
```

Comment:
`!smart_proxy.pulp_primary?` is redundant since `with_mirror_adapter` prevents `repository_mirror.rb` code from talking to the Pulp primary smart proxy.

Reply:
Fixed!